spec/controllers/steps_controller_inheritance_spec.rb

```diff
@@ -0,0 +1,104 @@
+require "rails_helper"
+
+RSpec.describe StepsController do
+  controller do
+    def update
+      @test = "in update"
+      head :no_content
+    end
+
+    def edit
+      @test = "in edit"
+      head :no_content
+    end
+
+    def index
+      @test = "in index"
+      head :no_content
+    end
+
+    private
+
+    def first_step_path
+      root_path
+    end
+  end
+
+  describe "#edit" do
+    before do
+      routes.draw { get "edit" => "steps#edit" }
+    end
+
+    context "when there is no application under way" do
+      it "redirects to the specified path" do
+        session[:snap_application_id] = nil
```

Comment:
should we have tests for `medicaid_application_id` as well?

Reply:
@jessieay good calllllll